lib/data/network/dio_cache_option.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:dio_cache_interceptor/dio_cache_interceptor.dart';
+import 'package:dio_cache_interceptor_hive_store/dio_cache_interceptor_hive_store.dart';
+import 'package:flutter/foundation.dart';
+import 'package:matrix/matrix.dart';
+import 'package:path_provider/path_provider.dart';
+
+class DioCacheOption {
+  static const String _hiveBoxName = "twake_dio_cache_hive_store";
+  static const Duration _maxStale = Duration(days: 7);
+
+  late HiveCacheStore _hiveCacheStore;
+  DioCacheOption._privateConstructor();
+
+  static DioCacheOption get instance => _instance;
+
+  static final DioCacheOption _instance = DioCacheOption._privateConstructor();
+
+  factory DioCacheOption() {
+    return _instance;
+  }
+
+  Future<String?> _getAppDirPath() async {
+    if (kIsWeb) return null;
+    final appDir = await getApplicationDocumentsDirectory();
```

Comment:
since you store cache files why not using `getTemporaryDirectory()` instead?

Reply:
<img width="500" alt="Screenshot 2023-09-18 at 23 31 20" src="https://github.com/linagora/twake-on-matrix/assets/99852347/17773f82-4633-45d3-b593-c245a65a2751">

1. Best practice  for init hive
2. `getTemporaryDirectory()` returns a directory for storing temporary files. It's typically used for caching and storing files that can be deleted when the app is not running.

However, for initializing Hive (a local NoSQL database), you would generally want to use a directory that persists across app launches. `getApplicationDocumentsDirectory()` is a more appropriate choice for initializing Hive because it provides a directory that persists across app launches and is meant for storing app-specific data.

Reply:
getApplicationDocumentsDirectory() will keep alive with the app until app uninstalled
getTemporaryDirectory() can be remove anytime
so I agree with Huy